hub/codec/codecs.py

```diff
@@ -0,0 +1,380 @@
+from io import BytesIO
+import numcodecs
+from numcodecs.abc import Codec
+
+import numpy as np
+from PIL import Image
+
+
+class Base:
+    """Base class for numcodec compressors"""
+
+    def encode(self, array: np.ndarray) -> bytes:
+        raise NotImplementedError()
+
+    def decode(self, bytes: bytes) -> np.ndarray:
+        raise NotImplementedError()
+
+
+class NumPy(Base):
+    """Numpy compressor"""
```

Comment:
What does this mean? Also, if you're going to capitalize the class like "NumPy" you should also refer to it as "NumPy"